test/NuGet.Clients.Tests/NuGet.PackageManagement.UI.Test/Models/PackageItemListViewModelTests.cs

```diff
@@ -37,5 +54,424 @@ public void LocalSources_PackageReader_NotNull()
+                {
+                    sourceStream.CopyTo(memoryStream);
+                    bytes = memoryStream.ToArray();
+                }
+                string grayiccImagePath = Path.Combine(testDir, "grayicc.png");
+                File.WriteAllBytes(grayiccImagePath, bytes);
+
+                var packageItemListViewModel = new PackageItemListViewModel()
+                {
+                    IconUrl = new Uri(grayiccImagePath, UriKind.Absolute)
+                };
+
+                // Act
+                BitmapSource result = await GetFinalIconBitmapAsync(packageItemListViewModel);
+
+                // Assert
+                VerifyImageResult(result, packageItemListViewModel.BitmapStatus);
+                Assert.Equal(IconBitmapStatus.DownloadedIcon, packageItemListViewModel.BitmapStatus);
+            }
+        }
+
+        [Fact]
+        public async Task IconUrl_WithValidImageUrl_FailsDownloadsImage_ReturnsDefault()
+        {
+            var iconUrl = new Uri("http://fake.test/image.png");
+
+            var packageItemListViewModel = new PackageItemListViewModel()
+            {
+                IconUrl = iconUrl
+            };
+
+            BitmapSource result = await GetFinalIconBitmapAsync(packageItemListViewModel);
+
+            VerifyImageResult(result, packageItemListViewModel.BitmapStatus);
+            Assert.Equal(IconBitmapStatus.DefaultIconDueToWebExceptionOther, packageItemListViewModel.BitmapStatus);
+        }
+
+        [LocalOnlyTheory]
+        [InlineData("icon.png", "icon.png", "icon.png", "")]
+        [InlineData("folder/icon.png", "folder\\icon.png", "folder/icon.png", "folder")]
+        [InlineData("folder\\icon.png", "folder\\icon.png", "folder\\icon.png", "folder")]
+        public async Task IconUrl_EmbeddedIcon_HappyPath_LoadsImage(
+            string iconElement,
+            string iconFileLocation,
+            string fileSourceElement,
+            string fileTargetElement)
+        {
+            using (var testDir = TestDirectory.Create())
+            {
+                // Create decoy nuget package
+                var zipPath = Path.Combine(testDir.Path, "file.nupkg");
+                CreateDummyPackage(
+                    zipPath: zipPath,
+                    iconName: iconElement,
+                    iconFile: iconFileLocation,
+                    iconFileSourceElement: fileSourceElement,
+                    iconFileTargetElement: fileTargetElement,
+                    isRealImage: true);
+
+                // prepare test
+                var builder = new UriBuilder(new Uri(zipPath, UriKind.Absolute))
+                {
+                    Fragment = iconElement
+                };
+
+                var packageItemListViewModel = new PackageItemListViewModel()
+                {
+                    IconUrl = builder.Uri,
+                    PackageReader = new Func<PackageReaderBase>(() => new PackageArchiveReader(zipPath))
```

Comment:
In all tests where you have a new `PackageArchiveReader`, make sure that all created instances are always disposed by the end of a test, both in success and failure cases.  Otherwise, directories get locked and don't get cleaned up correctly.

Reply:
line 552 of packageitemlistviewmodel.cs disposes the packagereader... 
using (PackageReaderBase reader = lazyReader())
so i think we are good.